File: app/src/main/java/com/openclassrooms/realestatemanager/Injection.java
package com.openclassrooms.realestatemanager;

import android.content.Context;

public class Injection {

    public static PhotoRepository providePhotoRepository(Context context) {
        AppDatabase appDatabase = AppDatabase.getInstance(context);
        return new PhotoRepository(appDatabase.photoDao());
    }

    public static RealEstateRepository provideRealEstateRepository(Context context) {
        AppDatabase appDatabase = AppDatabase.getInstance(context);
        return new RealEstateRepository(appDatabase.realEstateDao());
    }

    public static ViewModelFactory provideViewModelFactory(Context context) {
        RealEstateRepository dataSourceRealEstate = provideRealEstateRepository(context);
        PhotoRepository dataSourcePhoto = providePhotoRepository(context);
        return new ViewModelFactory(dataSourceRealEstate,dataSourcePhoto);
    }
}

File: app/src/main/java/com/openclassrooms/realestatemanager/injection/Injection.kt
package com.openclassrooms.realestatemanager.injection

import android.content.Context
import com.openclassrooms.realestatemanager.db.RealEstateDatabase
import com.openclassrooms.realestatemanager.repository.PhotoRepository
import com.openclassrooms.realestatemanager.repository.RealEstateRepository

object Injection {

    private fun provideRealEstateRepository(context: Context): RealEstateRepository {
        val db: RealEstateDatabase = RealEstateDatabase.getInstance(context)
        return RealEstateRepository(db.realEstateDao())
    }

    private fun providePhotoRepository(context: Context): PhotoRepository {
        val db: RealEstateDatabase = RealEstateDatabase.getInstance(context)
        return PhotoRepository()
    }

    fun provideViewModelFactory(context: Context): ViewModelFactory {
        val realEstateRepository: RealEstateRepository = provideRealEstateRepository(context)
        val photoRepository: PhotoRepository = providePhotoRepository(context)
        return ViewModelFactory(realEstateRepository,photoRepository)
    }
}